lib/backend/k8s/k8s_test.go

```diff
@@ -1297,6 +1310,131 @@ var _ = testutils.E2eDatastoreDescribe("Test Syncer API for Kubernetes backend",
 		})
 	})
 
+	// There are several states that we consider "finished", run the test for each one.
+	for _, finishPhase := range []k8sapi.PodPhase{k8sapi.PodSucceeded, k8sapi.PodFailed} {
```

Comment:
I'm not sure if it is really necessary but the "finished" check actually has 3 phases that it considers finished, with the 3rd being `podCompleted` though as commented is not part of the K8s API. I think we should include that in the tests especially since it is not actually from the K8s API.